openff/toolkit/topology/topology.py

```diff
@@ -1958,41 +1959,50 @@ def to_file(
             like PDB that truncate long atom names.
 
         """
-        from openff.units.openmm import to_openmm as to_openmm_quantity
-        from openmm import app
-        from openmm import unit as openmm_unit
-
-        # Convert the topology to OpenMM
-        openmm_top = self.to_openmm(ensure_unique_atom_names=ensure_unique_atom_names)
-
-        # Get positions in OpenMM format
-        if isinstance(positions, openmm_unit.Quantity):
-            openmm_positions = positions
-        elif isinstance(positions, unit.Quantity):
-            openmm_positions = to_openmm_quantity(positions)
-        elif isinstance(positions, np.ndarray):
-            openmm_positions = openmm_unit.Quantity(positions, openmm_unit.angstroms)
+        if positions is None:
+            positions = self.get_positions()
+
+        if isinstance(positions, np.ndarray):
+            positions = positions * unit.angstrom
         elif positions is None:
-            openmm_positions = to_openmm_quantity(self.get_positions())
-        else:
-            raise ValueError(f"Could not process positions of type {type(positions)}.")
+            raise ValueError(
+                "Positions must be specified in either molecule conformers or"
+                + " the positions argument."
+            )
+
+        if file_format.upper() == "PDB":
+            import openmm
+
+            # Convert the topology to OpenMM
+            openmm_top = self.to_openmm(
+                ensure_unique_atom_names=ensure_unique_atom_names
+            )
 
-        # Make sure the desired file format is PDB
-        if file_format.upper() != "PDB":
-            raise NotImplementedError("Topology.to_file supports only PDB format")
+            # Write PDB file
+            ctx_manager: Union[nullcontext[TextIO], TextIO]  # MyPy needs some help here
+            if isinstance(file, (str, Path)):
+                ctx_manager = open(file, "w")
+            else:
+                ctx_manager = nullcontext(file)
+            with ctx_manager as outfile:
+                openmm.app.PDBFile.writeFile(
+                    topology=openmm_top,
+                    positions=ensure_quantity(positions, "openmm"),
+                    file=outfile,
+                    keepIds=keep_ids,
+                )
+        elif file_format.upper() in ["SDF", "MOL2"]:
+            from rdkit.Chem import SDWriter
```

Comment:
Hmm, I don't like the direct import of RDKit in `topology.py` - The ToolkitRegistry system is supposed to keep direct imports out of this file. Could this be tucked into a private method in `rdkit_wrapper.py` instead?